src/datapoint.jl

```diff
@@ -0,0 +1,60 @@
+## New Type: DataPoint
+# Struct for carrying statistic and systematic uncertainty of a DataPoint
+struct DataPoint{T<:Number} <: Number
+    val::T
+    stat_err::Tuple{T,T}
+    sys_err::Tuple{T,T}
+end
+
+# Helper functions
+_to_tuple(x::T where T<:Number) = (x,x)
+_to_tuple(x::Tuple) = promote(x...)
+
+## Constructors
+# default constructor
+DataPoint() = DataPoint(NaN, (NaN, NaN), (NaN, NaN))
+# dispatch constructor
+DataPoint(x::Number, stat, sys = zero(x)) = DataPoint(x, stat = stat, sys = sys)
+# keyword constructor for integers (ceils uncertainties)
+function DataPoint(x::Integer; stat=zero(x), sys=zero(x))
```

Comment:
The more I think about the less I think we should special-case integer values and we should just promote then `x` to a floating point type:

Having an integer mean and non-integer uncertainty is meaningful and actually more insightful. Consider the following sets of integer data :
`(4,4,4,4,...,4,4,4)`
`(4,4,4,4,...,4,3,5)`
`(3,5,3,5,...,5,3,5)`
They all have the same mean but using the current implementation the second and third set would also have the same standard deviation which seems misleading since the standard deviation of the second set can get arbitrarily close to zero.

Reply:
Hmm you are right, this is a very delicate issue.
But for now I guess it is better to promote to floating point type.